docs/src/guides/smartweave/warp/deploying-contracts.md

```diff
@@ -0,0 +1,106 @@
+:::
+
+## Examples
+
+::: tip
+By default all deploy functions are published to Arweave via Bundlr-Network, each option has a flag that can be set to not use Bundlr, but it can take many confirmations for the network to fully confirm the transaction.
+:::
+
+**deploy**
+
+Deploys contract plus source code to Warp Sequencer, to Bundlr (L2), to Arweave.
+
+```ts
+const { contractTxId, srcTxId } = await warp.createContract.deploy({
+  wallet,
+  initState,
+  data: { 'Content-Type': 'text/html', body: '<h1>Hello World</h1>' },
+  src: contractSrc,
+  tags
+})
+```
+
+* wallet - should be a wallet.json object
+* initState - is a stringified JSON object
+* data - is optional if you want to write data as part of your deployment
+* src - is the string or Uint8Array value of the source code for the contract
+* tags - is an array of name/value objects `{name: string, value: string}[]`
+
+**deployFromSourceTx**
+
+Already have the source on the permaweb? Then deployFromSourceTx is your tool of choice! With the permaweb you never have to worry about data changing so re-using source code for contracts is a no brainer.
+
+```ts
+const { contractTxId, srcTxId } = await warp.createContract.deployFromSourceTx({
+  wallet,
+  initState,
+  srcTxId: 'SRC_TX_ID'
+})
+```
+
+**deployBundled**
+
+Uses Warp Gateway Sequencer's endpoint to upload a raw data item to Bundlr and index it.
+
+```ts
+import { createData } from 'arbundles'
+
+const dataItem = createData(
+  JSON.stringify({
+    "manifest": "arweave/paths",
+    "version": "0.1.0",
+    "index": {
+      "path": "index.html"
+    },
+    "paths": {
+      "index.html": {
+        "id": "cG7Hdi_iTQPoEYgQJFqJ8NMpN4KoZ-vH_j7pG4iP7NI"
+      }
+    }
+  })
+  , { tags: [{'Content-Type': 'application/x.arweave-manifest+json' }]})
+const { contractTxId } = await warp.createContract.deployBundled(dataItem.getRaw());
+```
+
+
+**register**
+
+Uses Warp Gateway Sequencer's endpoint to index a contract that has been uploaded with bundlr.
+
+```ts
+import Bundlr from '@bundlr-network/client'
+
+const bundlr = new Bundlr('https://node2.bundlr.network', 'arweave', wallet)
+const { id } = await bundlr.upload('Some Awesome Atomic Asset',  { 
+  tags: [{'Content-Type': 'text/plain' }]
+})
+const { contractTxId } = await warp.createContract.register(id, 'node2') 
+```
+
+## Summary
+
+Why are there so many options to deploy contracts? To reduce duplication, why deploy the same source code for each contract you deploy, also a concept called Atomic Assets. The permaweb is very unique in its architecture, it provides a feature where you can deploy both digital data and the contract to manage that data generating the same transaction identifier. The result is dynamic data paired with an immutable set of data. Deploying contracts is just one piece of the Warp SDK, to learn more keep reading this guide!
```

Comment:
second sentence kind of doesnt make sense to me as an answer to the question "why are there so many options to deploy contracts?". Avoiding duplication makes sense, but then we just throw in "and also atomic assets" which doesnt really help in the readers understanding. 

how about "These methods exist to reduce duplication, enable advanced contract interactions, and allow for flexibility for testing and usage of the smartweave protocol."